node/src/server/rpc.rs

```diff
@@ -368,6 +381,46 @@ mod rpc_impl {
         fn get_logs(&self, filter_object: FilterObject) -> RpcResult<Vec<LogObject>>;
     }
 
+    #[derive(Serialize, Clone)]
+    pub enum Value {
+        Block(alloy::rpc::types::Block),
+    }
+
+    #[derive(Serialize, Clone)]
+    pub enum RpcValueOrNodes {
+        Value(Value),
+        Nodes(Vec<NodeId>),
+    }
+
+    #[rpc(server, namespace = "light")]
+    pub trait Light {
+        #[method(name = "ping")]
+        async fn ping(&self, node_id: NodeId) -> RpcResult<()>;
+
+        #[method(name = "store")]
+        async fn store(
```

Comment:
The Node ID matching should be done at the rpc level (with a helper function) and the actual message call at a lower level (in the light client)